### PR TITLE
Remove old-style memory management in CUDA code

### PR DESCRIPTION
**Pull request by stephenswat**
Like with the core CPU code, we still have a bit of the old-style CUDA memory management code in the repository which is now unused. This commit removes some of that code in the name of cleanup.

### CODE REVIEW

File: tests/cuda/test_cuda_containers.cpp [hunk 1]
```diff
@@ -11,7 +11,6 @@
 #include "vecmem/allocators/allocator.hpp"
 #include "vecmem/containers/vector.hpp"
 #include "vecmem/memory/memory_manager.hpp"
```
Comment (krasznaa) [inline tests/cuda/test_cuda_containers.cpp:13]: I think you could remove the "memory_manager.hpp" include as well. We can't remove that header until **all** of the "old style" memory management is removed, but at least this test should not include it anymore.

**Review by krasznaa — COMMENTED** [review]
Apart from the tiny suggestion, looks good.

File: tests/cuda/test_cuda_containers.cpp (hunk 1 above)
Comment (stephenswat) [reply tests/cuda/test_cuda_containers.cpp]: Good one, fixed!

**Review by krasznaa — APPROVED** [review]
Looks good.

**Pull request merged**